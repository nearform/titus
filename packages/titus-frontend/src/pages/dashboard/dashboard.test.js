import React from 'react'
import Dashboard from '.'
import { render, cleanup, fireEvent } from '@testing-library/react'

describe('<Dashboard />', () => {
  afterEach(cleanup)

  it('should render correctly', () => {
    const { asFragment } = render(<Dashboard />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should trigger button correctly', async () => {
    const { asFragment, getByText } = render(<Dashboard />)
    const button = getByText('LOGOUT')
    fireEvent.click(button)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should trigger href correctly', async () => {
    const { asFragment, getByText } = render(<Dashboard />)
    const href = getByText('Check out the docs')
    fireEvent.click(href)

    expect(asFragment()).toMatchSnapshot()
  })
})
