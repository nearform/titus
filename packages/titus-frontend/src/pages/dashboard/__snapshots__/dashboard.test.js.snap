// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="logout button"
    >
      LOGOUT
    </button>
    <img
      alt="Titus logo"
      src="img/Accel_Logo_Titus.svg"
      style="width: 100%; height: 100%; margin-bottom: 10vh; margin-top: 10vh;"
    />
    <p>
      Develop and Deploy to features quickly using Titus, an Accelerated Development & Deployment Stack. Titus is production ready and can be deployed to all major cloud providers.
    </p>
    <a
      href="https://nf-titus.netlify.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Check out the docs
    </a>
  </div>
</DocumentFragment>
`;

exports[`<Dashboard /> should trigger button correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="logout button"
    >
      LOGOUT
    </button>
    <img
      alt="Titus logo"
      src="img/Accel_Logo_Titus.svg"
      style="width: 100%; height: 100%; margin-bottom: 10vh; margin-top: 10vh;"
    />
    <p>
      Develop and Deploy to features quickly using Titus, an Accelerated Development & Deployment Stack. Titus is production ready and can be deployed to all major cloud providers.
    </p>
    <a
      href="https://nf-titus.netlify.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Check out the docs
    </a>
  </div>
</DocumentFragment>
`;

exports[`<Dashboard /> should trigger href correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="logout button"
    >
      LOGOUT
    </button>
    <img
      alt="Titus logo"
      src="img/Accel_Logo_Titus.svg"
      style="width: 100%; height: 100%; margin-bottom: 10vh; margin-top: 10vh;"
    />
    <p>
      Develop and Deploy to features quickly using Titus, an Accelerated Development & Deployment Stack. Titus is production ready and can be deployed to all major cloud providers.
    </p>
    <a
      href="https://nf-titus.netlify.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Check out the docs
    </a>
  </div>
</DocumentFragment>
`;
